File: CODE/gjk-ui/src/const/crud/libs/software.js
import { DH_NOT_SUITABLE_GENERATOR } from "constants";

/*
 *    Copyright (c) 2018-2025, inforbus All rights reserved.
 *
 * Redistribution and use in source and binary forms, with or without
 * modification, are permitted provided that the following conditions are met:
 *
 * Redistributions of source code must retain the above copyright notice,
 * this list of conditions and the following disclaimer.
 * Redistributions in binary form must reproduce the above copyright
 * notice, this list of conditions and the following disclaimer in the
 * documentation and/or other materials provided with the distribution.
 * Neither the name of the inforbus.com developer nor the names of its
 * contributors may be used to endorse or promote products derived from
 * this software without specific prior written permission.
 * Author: inforbus
 */

export const tableOption = {
  border: true,
  index: true,
  indexLabel: '序号',
  stripe: true,
  menuAlign: 'center',
  align: 'center',
  editBtn: false,
  delBtn: false,
  addBtn: false,
  column: [
	  // {
    //   label: '主键id',
    //   prop: 'id'
    // },
    {
      label: '软件框架库名',
      prop: 'softwareName'
    },
	  {
      label: '版本',
      prop: 'version',
      slot: true,
    },
	  {
      label: '文件路径',
      prop: 'filePath'
    },
	  {
      // label: '审批状态0：未审批，1：已审批',
      label: '审批状态',
      prop: 'applyState',
      dicData: [{
        label: '未处理',
        value: "0"
      }, {
        label: '已申请',
        value: "1"
      }, {
        label: '已通过',
        value: "2"
      }, {
        label: '已驳回',
        value: "3"
      }, {
        label: '驳回再申请',
        value: "4"
      }]
    },
    {
      // label: '审批状态0：未审批，1：已审批',
      label: '审批描述',
      prop: 'applyDesc'
    },
	  {
      label: '创建时间',
      prop: 'createTime'
    },
	  // {
    //   label: '修改时间',
    //   prop: 'updateTime'
    // },
	  {
      label: '描述',
      prop: 'description'
    },
	  // {
    //   label: '删除标识0-正常，1-删除',
    //   prop: 'delFlag'
    // },
  ]
}

File: CODE/gjk/gjk-project/gjk-project-api/src/main/java/com/inforbus/gjk/pro/api/entity/Software.java
/*
 *    Copyright (c) 2018-2025, inforbus All rights reserved.
 *
 * Redistribution and use in source and binary forms, with or without
 * modification, are permitted provided that the following conditions are met:
 *
 * Redistributions of source code must retain the above copyright notice,
 * this list of conditions and the following disclaimer.
 * Redistributions in binary form must reproduce the above copyright
 * notice, this list of conditions and the following disclaimer in the
 * documentation and/or other materials provided with the distribution.
 * Neither the name of the inforbus.com developer nor the names of its
 * contributors may be used to endorse or promote products derived from
 * this software without specific prior written permission.
 * Author: inforbus
 */
package com.inforbus.gjk.pro.api.entity;

import com.baomidou.mybatisplus.annotation.TableId;
import com.baomidou.mybatisplus.annotation.TableName;
import com.baomidou.mybatisplus.extension.activerecord.Model;
import lombok.Data;
import lombok.EqualsAndHashCode;

import java.time.LocalDateTime;

/**
 * 软件框架库表
 *
 * @author pig code generator
 * @date 2019-06-14 09:46:48
 */
@Data
@EqualsAndHashCode(callSuper = true)
@TableName("gjk_software")
public class Software extends Model<Software> {
private static final long serialVersionUID = 1L;

    /**
   * 主键id
   */
    @TableId
    private String id;

    /**
     *用户id
     */
    private int userId;
    
    /**
     * 软件框架库名字
     */
    private String softwareName;
    /**
   * 版本
   */
    private Double version;
    /**
   * 文件路径
   */
    private String filePath;
    /**
   * 审批状态0：未审批，1：已审批
   */
    private String applyState;
    /**
     * 审批状态0：未审批，1：已审批
     */
      private String applyDesc;
    /**
   * 创建时间
   */
    private LocalDateTime createTime;
    /**
   * 修改时间
   */
    private LocalDateTime updateTime;
    /**
   * 描述
   */
    private String description;
    /**
   * 删除标识0-正常，1-删除
   */
    private String delFlag;
  
	
    
}
